### ISSUE TITLE
Part: Refine breaks intersection curves

### ISSUE THREAD

**Issue by RiccaDS** [FreeCAD/FreeCAD#18058]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

I have experienced this issue many times during a project. What happens is that "refine" breaks the model very often for example at intersection of features, for example when two tubes intersect or when a tube and and additive pipe intersect and other similar situations. It doesn't seem to be able to manage curves on surfaces. Disabling refine on all features solves the problem despite not having clean surfaces anymore. 
![image](https://github.com/user-attachments/assets/5cd53ccf-1c9d-4d3d-9dc7-d26931e99ce8)

I attach an example file that has refine enabled. 
You can use the "check geometry" tool in the part design WB to  see lots of "Boolean operation: invalid curve on surface".
[refine.zip](https://github.com/user-attachments/files/17863439/refine.zip)
This holds true also for lastest devs.

### Full version info

```shell
OS: Ubuntu 24.10 (ubuntu:GNOME/ubuntu/xcb)
Architecture: x86_64
Version: 1.0.0.39109 (Git) Conda AppImage
Build type: Release
Branch: (HEAD detached at 1.0.0)
Hash: 2fcc5317fe3aee96ca73475986a577719fc78e20
Python 3.11.9, Qt 5.15.13, Coin 4.0.3, Vtk 9.2.6, OCC 7.7.2
Locale: English/United States (en_US)
Stylesheet/Theme/QtStyle: unset/FreeCAD Classic/Qt default
Installed mods: 
  * Curves 0.6.51
  * Estimate 0.1.2
  * fasteners 0.5.29
  * A2plus 0.4.68
```

### Subproject(s) affected?

None

### Anything else?

Here is the project I was working on and had plenty of these issues.
https://forum.freecad.org/viewtopic.php?p=793207#p793207

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by tarman3** [comment]
Check geometry, no errors.
Please write steps to reproduce.

**Comment by RiccaDS** [comment]
Just open file and in part design workbench check geometry after selecting pad001
Are the check geometry settings the same? 
![image](https://github.com/user-attachments/assets/6864015f-2bcc-473b-88b8-a5289a35a7d4)

**Comment by tarman3** [comment]
**Run boolean operation check** сontains an interesting description.
The opinion of a knowledgeable person is required.

**Comment by RiccaDS** [comment]
Yes I think that option possibly wasn't enabled by default and maybe I enabled it some time ago. Maybe it is not important to have it enabled. Don't know.

**Comment by maxwxyz** [comment]
@AIRCAP could you take a look at this?

**Comment by luzpaz** [comment]
bumping for visibility

**Comment by viktak** [comment]
I also ran into the same issue. The part slices and prints just fine though.

**Comment by maxwxyz** [comment]
@pieterhijma is this OCC related?
